src/components/web3/SendCryptoPopup.tsx

```diff
@@ -0,0 +1,663 @@
+            <div css={{ margin: "10px" }}>
+              <Text variant="body">{`${params.senderDisplayName} wants to send you ${params.amount} ${params.token}! Your current address is ${web3Address}.`}</Text>
+            </div>
+            <div css={actionsCSS}>
+              <button onClick={returnSIWEToOrigin} css={buttonCSS}>
+                Accept
+              </button>
+              <button onClick={rejectIncomingRequest} css={buttonCSS}>
+                Reject
+              </button>
+            </div>
+          </div>
+        </div>
+      )}
+    </div>
+  );
+};
+
+interface CryptoSendPopupProps {
+  outgoingRequests: CryptoTransactionParams[];
+  setOutgoingRequests: React.Dispatch<
+    React.SetStateAction<CryptoTransactionParams[]>
+  >;
+  web3Address: string;
+  jitsi: IJitsiMeetApi;
+  pending: string;
+  setPending: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const CryptoSendPopup: React.FC<CryptoSendPopupProps> = ({
+  outgoingRequests,
+  setOutgoingRequests,
+  web3Address,
+  jitsi,
+  pending,
+  setPending,
+}) => {
+  const [showing, setShowing] = useState(DEBUG_SHOW);
+  const [value, setValue] = useState(0);
+  const [pendingName, setPendingName] = useState("");
+  const sendAndRegisterOutgoingRequest = async () => {
+    const amount = value;
+    const token = "BAT";
+    const nonce = await getNonce();
+    const recipientDisplayName = (await jitsi.getRoomsInfo()).rooms
+      .filter((r: any) => r.isMainRoom)[0]
+      .participants.filter((p: any) => p.id === pending)[0].displayName;
+
+    const sendParams: CryptoTransactionParams = {
+      senderDisplayName: "me",
+      sender: "me",
+      amount,
+      token,
+      nonce,
+      recipient: pending,
+      recipientDisplayName,
+    };
+    const msg: CryptoSendMessage = {
+      type: "crypto",
+      msgType: "REQ",
+      payload: sendParams,
+    };
+    setOutgoingRequests([...outgoingRequests, sendParams]);
+
+    jitsi.executeCommand(
+      "sendEndpointTextMessage",
+      pending,
+      JSON.stringify({
+        type: "crypto",
+        msgType: "REQ",
+        payload: { amount, token, nonce },
+      })
+    );
+
+    setPending("");
+  };
+  useEffect(() => {
+    const _setName = async (id: string) => {
+      setPendingName(await getNameFromId(jitsi, id));
+    };
+    if (pending) {
+      _setName(pending);
+      setShowing(true);
+    } else {
+      setShowing(false);
+    }
+  }, [pending]);
+
+  return (
+    <div>
+      {showing && (
+        <div css={[popupBaseCSS, popupCSSBottomRight]}>
+          <div css={popupHeaderCSS}>Send Crypto</div>
+          <Divider />
+          <div css={popupContentCSS}>
+            <div css={{ fontSize: "16px" }}>
+              <span>Send: </span>
+              <input
+                type="text"
+                pattern="[0-9]*"
+                value={value}
+                css={inputCSS}
+                onChange={(e) => setValue(parseInt(e.target.value, 10))}
```

Comment:
Come to think of it, we mustn't use the javascript Number type:

```js
> parseInt("")
NaN
> parseInt("12345678901234567890")
12345678901234567000  // missing 890
> parseInt("123456789012345678901234567890")
1.2345678901234568e+29
> parseInt("1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890")
Infinity
> Number.MAX_SAFE_INTEGER
9007199254740991
```

Perhaps better to use BigInt:

```js
> BigInt("")
0n
> BigInt("12345678901234567890")
12345678901234567890n
> BigInt("123456789012345678901234567890")
123456789012345678901234567890n
> BigInt("1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890")
1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890n
> BigInt("what?")
// Uncaught SyntaxError: Cannot convert what? to a BigInt
> BigInt("0b10110")
22n
> BigInt("0XFF").toString()
'255'
> JSON.stringify(BigInt("1"))
// Uncaught TypeError: Do not know how to serialize a BigInt
> JSON.stringify(BigInt("1").toString())
'"1"'
```

Reply:
As a general response: We can note that the "units" in the input is actual *whole* units, not wei. For obvious reasons -- we dont want people to have to type 1e18 to send 1 BAT. I have gone ahead and addressed this by changing the input to a float type, and also there is a maximum allowed input (`Number.MAX_SAFE_INTEGER`). It seems hacky to just cap the send amount, but considering that (according to google) the max supply of BAT is 1.5B, it should be alright. Ethereum (for the case of futureproofing) has infinite supply, but theres ~100M in circulation at the moment, so this shouldn't be an issue in the future if there are expansions to the sending mechanism.

Reply:
Hmmm OK, you're right I wasn't considering the wei at all.
However, both `parseInt` and `parseFloat` produce the same return type of `Number` so suffer from the same imprecision. Number is just not very suitable for dealing with precise amounts of money or big numbers or small numbers or combinations of big and small numbers...

```js
> parseFloat("100.000000000000007")  // 100000000000000007000 wei
100  // 7000 wei lost
```

Not the end of the world 💸 , but we probably should be using a combination of `parseUnits` and `formatUnits` as described in https://docs.ethers.org/v5/api/utils/display-logic/ to

> convert between user-friendly strings (usually displaying ~ether~BAT) and the machine-readable values that contracts and maths depend on (usually in wei)

Reply:
651df37ff44c8c4e31cd9653c98e6e79bd4412d7 changes so that the transaction amount is a `string` until the call to `sendCrypto`, at which point we use `parseUnits` from ethers. There is a check on the outgoing request creation process to make sure that `amount` is a valid string, though I guess adding more try-catch's down the line couldn't hurt.